### PR TITLE
[Form] Fixed keeping hash of equal \DateTimeInterface on submit

### PR DESCRIPTION
**Pull request by HeahDude**
| Q             | A
| ------------- | ---
| Branch?       | 2.8
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | ~
| License       | MIT
| Doc PR        | ~

### CODE REVIEW

File: src/Symfony/Component/Form/Tests/Extension/Core/DataMapper/PropertyPathMapperTest.php [hunk 1]
```diff
@@ -357,4 +357,28 @@ public function testMapFormsToDataIgnoresDisabled()
 
         $this->mapper->mapFormsToData(array($form), $car);
     }
+
+    public function testMapFormsToDataDoesNotChangeEqualDateTimeInstance()
```
Comment (xabbuh) [inline src/Symfony/Component/Form/Tests/Extension/Core/DataMapper/PropertyPathMapperTest.php]: This test would have passed before, right? We should probably add one making use of `DateTimeImmutable`.

Comment (HeahDude) [reply src/Symfony/Component/Form/Tests/Extension/Core/DataMapper/PropertyPathMapperTest.php]: Done, but this can be simplified back in 3.4.

**Review by xabbuh — APPROVED** [review]

File: src/Symfony/Component/Form/Extension/Core/DataMapper/PropertyPathMapper.php [hunk 2]
```diff
@@ -73,16 +73,17 @@ public function mapFormsToData($forms, &$data)
             // Write-back is disabled if the form is not synchronized (transformation failed),
             // if the form was not submitted and if the form is disabled (modification not allowed)
             if (null !== $propertyPath && $config->getMapped() && $form->isSubmitted() && $form->isSynchronized() && !$form->isDisabled()) {
+                $propertyValue = $form->getData();
                 // If the field is of type DateTime and the data is the same skip the update to
```
Comment (OskarStark) [inline src/Symfony/Component/Form/Extension/Core/DataMapper/PropertyPathMapper.php]: DateTime or DateTimeInterface

Please adjust the comment, thank you

**Review by OskarStark — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Form/Extension/Core/DataMapper/PropertyPathMapper.php (hunk 2 above)
Comment (HeahDude) [reply src/Symfony/Component/Form/Extension/Core/DataMapper/PropertyPathMapper.php]: Done, thanks!

**Review by OskarStark — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @HeahDude.

**Pull request merged**